flytekit/core/type_engine.py

```diff
@@ -437,7 +447,29 @@ def to_literal_type(cls, python_type: Type) -> LiteralType:
         Converts a python type into a flyte specific ``LiteralType``
         """
         transformer = cls.get_transformer(python_type)
-        return transformer.get_literal_type(python_type)
+        res = transformer.get_literal_type(python_type)
+        data = None
+        if typing.get_origin(python_type) is typing.Annotated:
```

Comment:
get_origin behavior is different between python 3.7/8 and 3.9.

In python3.8
```
In [1]: from typing_extensions import Annotated

In [2]: class A():
   ...:     ...
   ...:

In [3]: ii = Annotated[int, A()]

In [4]: import typing

In [5]: typing.get_origin(ii)
Out[5]: int
```

In 3.9
```
>>> import typing
>>> class A():
...   ...
...
>>> ii = typing.Annotated[int, A()]
>>> typing.get_origin(ii)
<class 'typing.Annotated'>
```
we need to handle both.

Reply:
typing.Annotated is only supported by 3.9
developers will only be able to use the core.FlyteAnnotation object with this version of python in my eye
https://www.python.org/dev/peps/pep-0593/

Reply:
Two things:
1. That PEP was implemented in the [typing_extensions](https://github.com/python/typing/tree/master/typing_extensions) module, which essentially means this is available to all python versions starting from 3.6.
2. In python 3.8 and below, the implementation of `get_origin` present in the typing module is not aware of `Annotated`, instead, `get_origin` present in the `typing_extensions` module does the right thing in all versions (including 3.9 and onwards).

tldrl: let's standardize on `typing_extensions` and not worry about compatibility between different versions.

Reply:
got it
makes sense and will push shortly